src/MarginTrading.SettingsService.Core/Domain/ScheduleConstraint.cs

```diff
@@ -1,11 +1,39 @@
 ﻿using System;
+using JetBrains.Annotations;
 
 namespace MarginTrading.SettingsService.Core.Domain
 {
-    public class ScheduleConstraint
+    [UsedImplicitly]
+    public class ScheduleConstraint : IEquatable<ScheduleConstraint>
     {
         public DateTime? Date { get; set; }
-        public DayOfWeek? DayOfWeek { get; set; }
+        public DayOfWeek? DayOfWeek { get; set; } 
         public TimeSpan Time { get; set; }
+        
+        public ScheduleConstraintType GetConstraintType()
+        {
+            if (Date == null && DayOfWeek == default)

```

Comment:
this is a little confusing, may be check both properties for null, default or .HasValue
like
`Date == null && DayOfWeek == null`
OR
`Date == default && DayOfWeek == default`